### ISSUE TITLE
AngularJS files fail to load in ngUpgrade tests on CI

### ISSUE THREAD

**Issue by pkozlowski-opensource** [angular/angular#28578]
`ngUpgrade` fail frequently on CI while trying to load AngularJS files. An example of such a job could be found here: https://circleci.com/gh/angular/angular/191135?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

I remember discussing this with @gkalpak and my understanding it that the way we load AngularJS files in the `ngUpgrade` tests is a particular one as we want to test with multiple AngularJS versions. I know that @gkalpak did many improvements in this area but sadly it seems that issues still occur.

I don't know if we can find a permanent solution here but if not maybe an idea would be to limit tests to just one AngularJS version (given that there is not much movement on the AngularJS side)?

**Comment by gkalpak** [comment]
I am waiting for #28219 to be merged (currently blocked on https://github.com/angular/angular/pull/28219#discussion_r254600003), to see if it helps with (diagnosing) the flakes. If that doesn't help, then we need to change the tests (either disable some browsers, or testing against one AngularJS version, or having a different bazel target for each AngularJS version).

**Comment by pkozlowski-opensource** [comment]
Another occurrence if it helps investigations: https://circleci.com/gh/angular/angular/201732?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by devversion** [comment]
I've modified the Saucelabs launcher slightly and gather the actual Selenium browser logs. Took me a while to hit a flaky job, but this seems to be more helpful than the current failure message.

```js
{ level: 'SEVERE',
  message:
   'http://localhost:9876/socket.io/?EIO=3&transport=polling&t=MbupHEw&sid=_eqH4aIA_u8JMnNdAAAH - Failed to load resource: net::ERR_PROXY_CONNECTION_FAILED',
  timestamp: 1552509908616,
  type: '' }
{ level: 'SEVERE',
  message:
   'http://localhost:9876/base/npm/node_modules/angular-mocks/angular-mocks.js - Failed to load resource: net::ERR_PROXY_CONNECTION_FAILED',
  timestamp: 1552509908616,
  type: '' }
{ level: 'SEVERE',
  message:
   'http://localhost:9876/socket.io/?EIO=3&transport=polling&t=MbupHk1&sid=_eqH4aIA_u8JMnNdAAAH - Failed to load resource: net::ERR_PROXY_CONNECTION_FAILED',
  timestamp: 1552509908616,
  type: '' }
{ level: 'SEVERE',
  message:
   'http://localhost:9876/socket.io/socket.io.js 6 WebSocket connection to \'ws://localhost:9876/socket.io/?EIO=3&transport=websocket&sid=rTuc0lmQu0G4xG4PAAAI\' failed: Connection closed before receiving a handshake response',
```
<sup>From: https://circleci.com/gh/angular/angular/242773</sup>

Just adding this for tracking here. I'm looking more into it. A potential workaround would be to just don't go through the proxy for these requests.

Also Saucelabs seemed to have similar issues with Android 5 and 6. See [here](https://wiki.saucelabs.com/download/attachments/48365737/Sauce%20Labs%20Documentation%20v1.0.pdf?version=1&modificationDate=1447890948000&api=v2)

**Comment by devversion** [comment]
https://github.com/angular/angular/commit/6f3052b799396592952a0d035872778c061da803 just landed. Please post here if you still see flaky `legacy-saucelabs-test` jobs related to that issue.

**Comment by devversion** [comment]
Seems to already happen in https://circleci.com/gh/angular/angular/255436?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link.

**Comment by devversion** [comment]
Some more information for tracking. I was able to hit a flake with Saucelabs video recording enabled.

https://assets.saucelabs.com/jobs/82e0872d4a1b42288b80c40c38b0a0f0/video.mp4

At 1:06 the internet connection shortly breaks.. meaning that the AngularJS scripts cannot be loaded.

Test link for tracking: https://app.saucelabs.com/tests/82e0872d4a1b42288b80c40c38b0a0f0

```js
{
    "source": "console-api",
    "message": "http://localhost:9876/context.js 252:44 \"WARNING: sanitizing unsafe URL value javascript:true (see http://g.co/ng/security#xss)\"",
    "timestamp": 1553616309158,
    "level": "WARNING"
},
{
    "source": "console-api",
    "message": "http://localhost:9876/context.js 252:44 \"WARNING: sanitizing unsafe URL value javascript:true (see http://g.co/ng/security#xss)\"",
    "timestamp": 1553616309158,
    "level": "WARNING"
},
{
    "source": "console-api",
    "message": "http://localhost:9876/context.js 252:44 \"WARNING: sanitizing unsafe URL value \\\"http://server\\\" (see http://g.co/ng/security#xss)\"",
    "timestamp": 1553616309159,
    "level": "WARNING"
},
{
    "source": "console-api",
    "message": "http://localhost:9876/context.js 252:44 \"WARNING: sanitizing unsafe style value url(\\\"http://server\\\") (see http://g.co/ng/security#xss).\"",
    "timestamp": 1553616309159,
    "level": "WARNING"
},
{
    "source": "network",
    "message": "http://localhost:9876/base/npm/node_modules/angular-1.6/angular.min.js - Failed to load resource: net::ERR_PROXY_CONNECTION_FAILED",
    "timestamp": 1553616309180,
    "level": "SEVERE"
},
{
    "source": "javascript",
    "message": "http://localhost:9876/socket.io/socket.io.js 6 WebSocket connection to 'ws://localhost:9876/socket.io/?EIO=3&transport=websocket&sid=cVuBZK5sNWB_p3buAAAE' failed: Error in connection establishment: net::ERR_NETWORK_CHANGED",
    "timestamp": 1553616309182,
    "level": "SEVERE"
}
```

**Comment by devversion** [comment]
I've updated the comment above to reflect even more findings. Looking into the Logcat device output, it looks like it is somehow related to networking/SIM insertion. Sent those findings to the Saucelabs team.

This should help them debugging the issues which only surface on mobile Android devices. The SIM emulation/network issue would then explain why we only see this on Android Saucelabs devices.

**Comment by gkalpak** [comment]
After #29603, I haven't seen another flake. I will close this as fixed, but please re-open/open a new issue, if you see more flakes.
